File: 936-stamping-the-sequence/936-stamping-the-sequence.cpp
class Solution {
public:
    
    void update(string& curr, int i, string& stamp){
        
        for(int j=0; j<stamp.length(); j++){
            curr[i+j] = '*';
        }
    }
    
    int findStampIndex(string& curr, string& stamp){
        
        for(int i=0; i<=curr.length()-stamp.length(); i++){
            
            int j=0; // for stamp
            int s = i; // for curr
            bool notStarChar = false;
            
            while(j < stamp.length() && s < curr.length() && (curr[s] == '*' || curr[s] == stamp[j])){
                if(curr[s] != '*') notStarChar = true;
                s++;
                j++;
            }
            
            if(j == stamp.length() and notStarChar){
                return i;
            }
        }
        
        return -1;
    }
    vector<int> movesToStamp(string stamp, string target) {
        
        vector<int> res;
        string curr = target;
        string targetStr(target.length(), '*');
        
        while(curr != targetStr){
            
            int stampIndex = findStampIndex(curr, stamp);
            if(stampIndex < 0){
                vector<int> v;
                return v;
            }
            else{
                update(curr, stampIndex, stamp);
            }
            
            res.push_back(stampIndex);
        }
        
        reverse(res.begin(), res.end());
        return res;
    }
};

File: 936-stamping-the-sequence/936-stamping-the-sequence.py
class Solution:
    
    def update(self, curr, i, stamp):
        for j in range(len(stamp)):
            curr[j+i] = "*"
    
    def findIndex(self, curr, stamp):
        
        for i in range(0, len(curr) - len(stamp) + 1):
            j, s = 0, i
            isNotStarChar = False
            while s<len(curr) and j<len(stamp) and (curr[s] == "*" or curr[s] == stamp[j]):
                if curr[s] != "*":
                    isNotStarChar = True
                s += 1
                j += 1
            
            if j == len(stamp) and isNotStarChar:
                return i
        
        return -1
    
        
    def movesToStamp(self, stamp: str, target: str) -> List[int]:
        
        curr, targetStr = list(target), list("*"*len(target))
        res = []
        
        while curr != targetStr:
            
            stampIdx = self.findIndex(curr, stamp)
            
            if stampIdx < 0:
                return []
            else:
                self.update(curr, stampIdx, stamp)
            
            res.append(stampIdx)
        
        res.reverse()
        return res
    
            